File: test-suite-kotlin/src/test/kotlin/io/micronaut/chatbots/basecamp/HelloWorldHandler.kt
package io.micronaut.chatbots.basecamp

import io.micronaut.chatbots.basecamp.api.Query
import io.micronaut.chatbots.basecamp.core.BasecampBotConfiguration
import io.micronaut.chatbots.basecamp.core.BasecampHandler
import io.micronaut.context.annotation.Requires
import jakarta.inject.Singleton
import java.util.Optional

@Requires(property = "spec.name", value = "basecamp")
// tag::class[]
@Singleton
class HelloWorldHandler : BasecampHandler {

    override fun canHandle(bot: BasecampBotConfiguration, input: Query): Boolean = true
    override fun handle(bot: BasecampBotConfiguration, input: Query): Optional<String> = Optional.of("Hello World")
}
// end::class[]

File: test-suite-python/src/test/python/micronaut/chatbots/docs/basecamp/HelloWorldHandler.py
from jakarta.inject import Singleton
from java.util import Optional
from micronaut.chatbots.basecamp.api import Query
from micronaut.chatbots.basecamp.core import BasecampBotConfiguration, BasecampHandler
from micronaut.context.annotation import Requires


@Requires(property="spec.name", value="basecamp")
# tag::class[]
@Singleton
class HelloWorldHandler(BasecampHandler):

    def canHandle(self, bot: BasecampBotConfiguration | None, input: Query) -> bool:
        return "hello" in input.getCommand()

    def handle(self, bot: BasecampBotConfiguration | None, input: Query) -> Optional[str]:
        return Optional.of("Hello World")
# end::class[]
